compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -1140,6 +1151,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     fn check_pat_path(
         &self,
         hir_id: HirId,
+        pat_id: HirId,
```

Comment:
can we name this something other than `hir_id` xd maybe `path_id`, i dont think its super clear immediately why we accept two hirids here or why